taxonomy/views.py

```diff
@@ -0,0 +1,79 @@
+"""
+Views for the taxonomy app
+"""
+
+from django.shortcuts import render, redirect
+
+from .forms import VocabularyForm
+from learningresources.models import Repository
+from taxonomy.models import Vocabulary
+
+
+def edit(request, vocabulary_id=None):
+    """
+    Show form to create a new vocabulary
+    """
+    # TODO: this function can be made simpler
+    if vocabulary_id is None:
+        form = VocabularyForm()
+    else:
+        vocabulary = Vocabulary.objects.get(id=vocabulary_id)
+        form = VocabularyForm(instance=vocabulary)
+
+    # TODO: where to specify this?
+    repository = Repository.objects.first()
+
+    form.instance.repository = repository
+    # TODO: are these good defaults?
+    form.instance.required = False
+    form.instance.weight = 1000
+
+    if request.POST:
+        form = VocabularyForm(request.POST)
+
+        form.instance.repository = repository
+        # TODO: are these good defaults?
+        form.instance.required = False
```

Comment:
The defaults should be in the model. I don't have a comment on whether these are good.